### ISSUE TITLE
Drop `v1_layers` in COLRv1?

### ISSUE THREAD

**Issue by drott** [googlefonts/fontations#632]
[v1_layer](https://docs.rs/read-fonts/0.11.1/read_fonts/tables/colr/type.Colr.html#method.v1_layer) is described as _"Returns the COLRv1 layer at the given index."_ - but it's not clear at which level in the paint graph this call is supposed to happen. 

I am suggesting to either add that documentation, or - preferarbly - drop this API method. I don't think we should talk about layers with COLRv1. A PaintColrLayers table can appear as the root of the graph, but it can also appear in other locations in the paint graph as a shortcut for a chain of PaintComposites. 

Since also, a PaintColrLayers interface exists that allows enumerating layers, it seems to be that v1_layer is redundant and - as described above - not 100% clear at which level it applies.

**Comment by drott** [comment]
Or wait - reading this again, it seems that the index values returned from PaintColrLayers are supposed to be passed to v1_layer? Is that the idea? In which case - I'd at least suggest to update the implementation.

The approach in the API may be closer to the font structure, but I'd find it more intuitive to be able to retrieve a Paint + PaintId object directly from the PaintColrLayers interface. This may make it easier to check that the right layers are retrieved. Just my 2 cents.

**Comment by dfrg** [comment]
Right, the intent is that you'd iterate over the indices in `ResolvedPaint::ColrLayers::range` and pass each to `Colr::v1_layer` to retrieve the paint and associated id. Seems simple enough to change this variant to hold enough context to provide an iterator interface to the layer list. I'll also improve the docs.
